pandas/tests/util/test_assert_produces_warning.py

```diff
@@ -178,6 +178,14 @@ def test_same_category_different_messages_last_match():
         warnings.warn("Do not match that either", category)
         warnings.warn("Match this", category)
 
+def test_match_multiple_warnings():
+    category = (FutureWarning, UserWarning)
+    with tm.assert_produces_warning(category, match=r"^Match this"):
+        warnings.warn("Match this",FutureWarning )
+        warnings.warn("Match this too", UserWarning)
+        warnings.warn("Do not match this", DeprecationWarning)
```

Comment:
I don't think you need this (the test would error because of an extra warning being caught)

Reply:
Should I add a test, what would the test for the annotation look like?

Reply:
No need to add tests for type annotations, you just need a tests checking that multiple warnings can be caught